### ISSUE TITLE
🐛 Astro project linting -  Illegal return statement outside of a function

### ISSUE THREAD

**Issue by morriq** [biomejs/biome#2032]
### Environment information

```block
CLI:
  Version:                      1.6.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.11.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.4"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

I have simple page and ts code in it:

```
---
const { isLoggedIn, redirectToLoggedInPage } = Astro.locals.session;

if (isLoggedIn()) {
  return redirectToLoggedInPage();
}
---
```

When I run biome lint ./ then the errors appear:

<img width="779" alt="Screenshot 2024-03-10 at 11 36 53" src="https://github.com/biomejs/biome/assets/2962338/9a50819b-f462-4e64-b85b-a098cbcbc6c9">

### Expected result

It should not throw error

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
You should either disable the rule for astro files or suppress it, at the moment Biome doesn't know that the frontmatter runs in its own scope

**Comment by kremalicious** [comment]
@ematipico but how to suppress this specifically? 

Like, this does not work as the rule is part of the parser itself:

```
---
if (!slug) {
  // biome-ignore parse: perfectly fine in Astro
  return null ←-- still errrors
}
---
```

I am using Biome 2.3.5 with experimental HTML support for Astro files
